common/src/main/java/org/apache/apex/common/util/JarHelper.java

```diff
@@ -186,4 +190,62 @@ private void addEntry(File file, String name) throws IOException
       jos.closeEntry();
     }
   }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file
+   * @param classPath Class path
+   * @return List of jar-files
+   */
+  public static List<String> getApexDependentJarsByClass(String classPath)
+  {
+    try {
+      return getApexDependentJarsByClass(Thread.currentThread().getContextClassLoader().loadClass(classPath));
+    } catch (ClassNotFoundException ex) {
+      logger.error("Cannot find the class {}", classPath, ex);
+      throw new RuntimeException("Cannot find the class " + classPath, ex);
+    }
+  }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file
+   * @param clazz Class
+   * @return List of jar-files
+   */
+  public static List<String> getApexDependentJarsByClass(Class clazz)
+  {
+    List<String> list = new LinkedList<>();
+
+    try {
+      URLConnection conn = clazz.getResource(clazz.getSimpleName() + ".class").openConnection();
```

Comment:
Unfortunately I cannot use the implementation of the method JarHelper.getJar(). The method getJar() creates jar file if the method parameter Java class does not belong to any jar-file. And it does not satisfy to the PR requirements (please talk to @PramodSSImmaneni about the PR requirements).

Logically the implementation of the method getJar() is equal to the following 2 lines of Java code:

    URLConnection conn = clazz.getResource(clazz.getSimpleName() + ".class").openConnection();
    return conn instanceof JarURLConnection ? ((JarURLConnection)conn).getJarFileURL().getFile() : null;

The method getJar() returns a name of jar-file as a string. But the implementation of the method getApexDependentJarsByClass() should continue to work with URLConnection object to retrieve the manifest properties.

I think the code will look very ugly, if after getting of the jar-file name we need to recreate the URLConnection object from string again. It does not make sense.

Also in order to use the method JarHelper.getJar() the code should create an instance of the class JarHelper. But logically the method getJar() should be static , because it does not depend on any environment. (The implementation of the method contains caching of jar-name that is not used by the implementation of the method getApexDependentJarsByClass()).